File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task104_CLAUDE_claude-3-5-sonnet-20240620.java

import java.util.Scanner;

public class BufferInput {
    private static final int BUFFER_SIZE = 20;

    public static void main(String[] args) {
        char[] buffer = new char[BUFFER_SIZE];
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter a string (max " + BUFFER_SIZE + " characters):");
        String input = scanner.nextLine();

        int length = Math.min(input.length(), BUFFER_SIZE);
        input.getChars(0, length, buffer, 0);

        System.out.println("Buffer contents:");
        System.out.println(new String(buffer, 0, length));

        scanner.close();
    }
}

File: dataset/output/CLAUDE_claude-3-5-sonnet-20240620/Task104_CLAUDE_claude-3-5-sonnet-20240620.c

#include <stdio.h>
#include <string.h>

#define BUFFER_SIZE 100

void handleUserInput(char* buffer, int bufferSize) {
    printf("Enter a string (max %d characters): ", bufferSize - 1);
    
    if (fgets(buffer, bufferSize, stdin) != NULL) {
        size_t len = strlen(buffer);
        if (len > 0 && buffer[len - 1] == '\\n') {
            buffer[len - 1] = '\\0';
        } else if (len == bufferSize - 1) {
            int c;
            while ((c = getchar()) != '\\n' && c != EOF);
            printf("Input too long. Truncated to %d characters.\\n", bufferSize - 1);
        }
    }

    printf("You entered: %s\\n", buffer);
}

int main() {
    char buffer[BUFFER_SIZE];
    handleUserInput(buffer, BUFFER_SIZE);
    return 0;
}
